### DISCUSSION TITLE
PnP.PowerShell : "Add-PnPSiteTemplate" command not working

### DISCUSSION THREAD

**Discussion by MAP-Git** [pnp/powershell#1184] [Q&A]
I want to execute Add-PnPSiteTemplate command but am not sure with property 'SiteTemplate' and 'TenantTemplate'. Which values are to be passed as parameters. 

Please assist how to execute this command successfully.

My requirement is : add Site Template to template 'from your Organization' section of SharePoint site.

Appreciate for help and guidance in advance. Thanks!

**Comment by MAP-Git** [comment]
@erwinvanhunen, please do needful here or contact me to someone who can help me here.

**Comment by KoenZomers** [comment]
You're using the wrong cmdlet to achieve that goal. You first should create a JSON schema. Documentation on this here: https://docs.microsoft.com/en-us/sharepoint/dev/declarative-customization/site-design-json-schema . You could use Get-PnPSiteScriptFromWeb or Get-PnPSiteScriptFromList to generate the JSON from an existing site or web. Then you need to use Add-PnPSiteScript to register this script followed by Add-PnPSiteDesign to create a site design out of the script. I've added a new cmdlet Add-PnPSiteDesignFromWeb through PR https://github.com/pnp/powershell/pull/1201 which will do all these steps in one cmdlet.

**Reply by MAP-Git** [reply]
Thanks for the reply and explanation @KoenZomers !

[](url
![Site-Template](https://user-images.githubusercontent.com/66069197/135888765-b77c3ca7-fb63-4e06-8c69-372540373c10.png)
)

Can you please tell, your solution would work for 'From your Orginization' section? Means Add-PnPSiteDesign cmdlet would add template to 'From your Organization'.

**Reply by MAP-Git** [reply]
Hi @KoenZomers, I checked it worked for 'From your Organization' section. 

One more question, using the above solution I can't add Site Pages and Site Assets library. **Please advise, how can we add Site Pages and Site Assets items as Site Template?**

**Reply by KoenZomers** [reply]
Indeed, the site design options are still limited to only these components:
![image](https://user-images.githubusercontent.com/5940670/135904989-478da2c6-a8e2-4c91-b4ca-896ee22954ec.png)

Source: https://docs.microsoft.com/en-us/sharepoint/dev/declarative-customization/site-design-json-schema

If you wish to be able to deploy more than this, you're entering a more complex scenario which isn't easy to explain and you will have many options to achieve it. What the best option would be depends on various factors. One option could be to use a site design, leverage the [trigger a flow](https://docs.microsoft.com/en-us/sharepoint/dev/declarative-customization/site-design-json-schema#trigger-a-flow) action to kick off a Power Automate Flow (requires a premium license for every user potentially being able to kick off the Flow) or a Logic App (pay per run, generally cheaper). From there you could execute either raw commands towards SharePoint and/or Graph to do what you want or connect to an Azure Runbook or Azure Function to run PnP PowerShell and use PnP Templates. This will open the door to do pretty much anything you would like, yet retaining the easy to use end user dialog for picking a template. It's technically a bit tricky to set up though, if you're not familiar with this yet. Documentation on the provisioning framework in PnP can be found here: https://docs.microsoft.com/en-us/sharepoint/dev/solution-guidance/introducing-the-pnp-provisioning-engine
